src/odemis/acq/stream/_projection.py

```diff
@@ -436,3 +530,617 @@ def _updateImage(self):
 
         except Exception:
             logging.exception("Updating %s %s image", self.__class__.__name__, self.name.value)
+
+
+class MeanSpectrumProjection(DataProjection):
+    """
+    Compute the global spectrum of the data as an average over all the pixels
+    returns (numpy.ndarray of float): average intensity for each wavelength
+     You need to use the metadata of the raw data to find out what is the
+     wavelength for each pixel, but the range of wavelengthBandwidth is
+     the same as the range of this spectrum.
+    """
+
+    def __init__(self, stream):
+        super(MeanSpectrumProjection, self).__init__(stream)
+        self.stream.calibrated.subscribe(self._on_new_spec_data)
+        self._shouldUpdateImage()
+
+    def _on_new_spec_data(self, _):
+        self._shouldUpdateImage()
+
+    # TODO: have an "area=None" argument which allows to specify the 2D region
```

Comment:
This TODO doesn't make sense here, as `updateImage()` is already defined, and cannot take extra arguments. I'd suggest to rephrase the TODO, and say it could be done via a .area VA (and move the comment to the `__init__`).